enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,731 @@
+4. Alice will now have a cluster that has been setup with workload pinning.
+
+#### Variation [optional]
+
+##### E2E Workflow deployment
+
+This section outlines an end-to-end workflow for deploying a cluster with
+workload partitioning enabled and how pods are correctly scheduled to run on the
+management CPU pool.
+
+> Note: Items in bold are modifications/additions from the previous enhancement,
+> [Management Workload Partitioning](management-workload-partitioning.md)
+
+1. User sits down at their computer.
+2. **The user creates a `PerformanceProfile` resource with the desired
+   `isolated` and `reserved` CPUSet with the `cpu.workloads[Infrastructure]`
+   added to the enum list.**
+3. **Alice updates the `Infrastructure` CR status to denote that workload
+   partitioning is turned on.**
+4. The user runs the installer to create the standard manifests, adds their
+   extra manifests from steps 2, then creates the cluster.
+5. The kubelet starts up and finds the configuration file enabling the new
+   feature.
+6. The kubelet advertises `management.workload.openshift.io/cores` extended
+   resources on the node based on the number of CPUs in the host.
+7. The kubelet reads static pod definitions. It replaces the `cpu` requests with
+   `management.workload.openshift.io/cores` requests of the same value and adds
+   the `resources.workload.openshift.io/{container-name}: {"cpushares": 400}`
+   annotations for CRI-O with the same values.
+8. **NTO will generate the machine config manifests and apply them.**
+9. **MCO modifies kubelet and cri-o configurations of the relevant machine pools
+   to the updated `reserved` CPU cores and restarts the nodes**
+10. Something schedules a regular pod with the
+    `target.workload.openshift.io/management` annotation in a namespace with the
+    `workload.openshift.io/allowed: management` annotation.
+11. The admission hook modifies the pod, replacing the CPU requests with
+    `management.workload.openshift.io/cores` requests and adding the
+    `resources.workload.openshift.io/{container-name}: {"cpushares": 400}`
+    annotations for CRI-O.
+12. The scheduler sees the new pod and finds available
+    `management.workload.openshift.io/cores` resources on the node. The
+    scheduler places the pod on the node.
+13. Repeat steps 10-12 until all pods are running.
+14. Cluster deployment comes up with management components constrained to subset
+    of available CPUs.
+
+##### Partition Resize workflow
+
+This section outlines an end-to-end workflow for resizing the CPUSet partition.
+
+> Note: Items in bold are modifications/additions from the previous enhancement,
+> [Management Workload Partitioning](management-workload-partitioning.md)
+
+1. User sits down at their computer.
+2. **The user updates the `PerformanceProfile` resource with the new desired
+   `isolated` and new `reserved` CPUSet with the `cpu.workloads[Infrastructure]`
+   in the enum list.**
```

Comment:
Can they edit an existing PerformanceProfile in the same way? If so, we don't need a completely separate workflow description for that but it would be good to point out.

Reply:
I am more worried about the enum and the handling of its modifications.

Reply:
After our discussion, it was decided that for this enhancement we would be changing the PerformanceProfile API